### PR TITLE
Add Delete verb to cache GRPC

### PR DESCRIPTION
**Pull request by kewilli**
[AB#1565544](https://dev.azure.com/mseng/708e929f-6bd5-415a-8daf-25b1dac08dd8/_workitems/edit/1565544)

### CODE REVIEW

**Comment by JuanCarlosGI** [comment]
Should we make a distinction between ContentNotDeleted and ContentNotFound? ContentNotDeleted could mean that there was an exception on the server side and so the operation could not be completed, and so we might want to retry, where ContentNotFound clearly talls the caller not to retry, and assume a success, even though deleted size will not be available

File: Public/Src/Cache/ContentStore/Grpc/ContentStore.proto [hunk 1]
```diff
@@ -199,6 +201,19 @@ message ExistenceResponse {
...
 
+// Delete
+message DeleteContentRequest {
+    string traceId = 1;
+    int32 hashType = 2;
+    bytes contentHash = 3;
+}
+
+message DeleteContentResponse {
+    ResponseHeader Header = 1;
```
Comment (dannyvv) [inline Public/Src/Cache/ContentStore/Grpc/ContentStore.proto]: >H [](start = 19, length = 1)

casing is off.
Protobuf recommended casing is camlCase. The C# codegen will make it PascalCase....

I see you just copied, but you should probably do a pass before too many folks take dependencies and clean it up.

**Comment by SergeyTeplyakov** [comment]
>                 ? $"Success Hash={ContentHash} Size={EvictedSize} Pinned={PinnedSize}"

Do you want to print the error code here?

---
Refers to: Public/Src/Cache/ContentStore/Interfaces/Results/DeleteResult.cs:94 in d587241. [](commit_id = d587241ec10e97b8279cf1f6a8213461344cfdda, deletion_comment = False)

File: Public/Src/Cache/ContentStore/Grpc/ContentStore.proto (hunk 1 above)
Comment (kewilli) [reply Public/Src/Cache/ContentStore/Grpc/ContentStore.proto]: I hope that isn't a dangerous change! I'll migrate this entire file to camelCase after this PR is checked in.

---
In reply to: [303659836](https://github.com/microsoft/BuildXL/pull/599#discussion_r303659836) [](ancestors = 303659836)

Comment (ljcollins25) [reply Public/Src/Cache/ContentStore/Grpc/ContentStore.proto]: I don't see a point in changing this. It's purely cosmetic and more likely to cause problems than it is to help.

---
In reply to: [303673571](https://github.com/microsoft/BuildXL/pull/599#discussion_r303673571) [](ancestors = 303673571,303659836)

**Pull request merged**